
package swgemurpcserver.rpc;

message GetAccountRequest {
	enum SearchType {
		ACCOUNTID = 0;
		ACCOUNTNAME = 1;
	}
	required SearchType search_type = 1;
	optional string user_name = 2;
	optional string password = 3;
	optional fixed32 account_id = 4;
}

message SWGEmuAccountCharacter {
	required fixed64 object_id = 1;
	required fixed32 account_id = 2;
	required fixed32 galaxy_id = 3;	
	required fixed64 create_time = 4;
	required fixed32 gender = 5;
	required fixed32 race = 6;
	required string first_name = 7;
	required string sur_name = 8;
	required string galaxy_name = 9;
			
	optional bool banned = 10 [ default = false ];
	optional fixed32 ban_expiration = 11;
	optional string ban_reason = 12;
	
	optional string inventory = 13;
}
message SWGEmuAccount {
	required fixed32 account_id = 1;
	required string user_name = 2;
	required bool active = 3;
	required fixed32 created_time = 4;
	optional fixed32 admin_level = 5;
	repeated  SWGEmuAccountCharacter characters = 10;
}

message GetAccountResponse {
	repeated SWGEmuAccount accounts = 1;
	optional bool valid_password = 2;
}

option cc_generic_services = true;

service SWGEmuAccountService {
	rpc GetAccount (GetAccountRequest) returns (GetAccountResponse);
}